packages/expo-sharing/ios/SharingModule.swift

```diff
@@ -0,0 +1,64 @@
+import ExpoModulesCore
+
+struct SharingContext {
+  let pendingPromise: Promise
+  let delegate: SharingDelegate
+}
+
+public final class SharingModule: Module, OnDocumentInteractionResult {
+  private var documentInteractionController: UIDocumentInteractionController?
+  private var sharingContext: SharingContext?
+
+  public func definition() -> ModuleDefinition {
+    Name("ExpoSharing")
+
+    AsyncFunction("shareAsync") { (url: URL, options: SharingOptions, promise: Promise) in
+      guard sharingContext == nil else {
+        throw SharingInProgressException()
+      }
+
+      guard let filePermissions: EXFilePermissionModuleInterface =
+        appContext?.legacyModule(implementing: EXFilePermissionModuleInterface.self)
+      else {
+        throw FilePermissionModuleException()
+      }
+
+      let grantedPermissions = filePermissions.getPathPermissions(url.absoluteString)
+      guard grantedPermissions.rawValue <= EXFileSystemPermissionFlags.read.rawValue else {
```

Comment:
I think it should be `>=` to pass when granted `read` or `write` permission 🤔 Now it passes for `none` too
```suggestion
      guard grantedPermissions.rawValue >= EXFileSystemPermissionFlags.read.rawValue else {
```

Reply:
I had it this way initially but the function always fails with you don't have access to the provided file We request the permissions so I'm not sure why it's happening 🤔